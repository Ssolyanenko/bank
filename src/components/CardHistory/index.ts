export * from './History';
export * from './Pagination';
export * from './CardHistory';
