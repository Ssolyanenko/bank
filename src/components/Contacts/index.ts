export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
