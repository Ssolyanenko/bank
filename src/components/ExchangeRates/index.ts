export * from './ExchangeRates';
export * from './Converter';
export * from './Rates';
