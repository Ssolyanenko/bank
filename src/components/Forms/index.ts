export * from './Login';
export * from './PasswordRecoveryForm';
export * from './GeneralInfo';
export * from './Security';
export * from './OrderDebitTravelCard';
export * from './SetPinCodeForm';
export * from './OrderCreditCardPremium';
export * from './FormNotifications';
export * from './FilterHistoryForm';
export * from './CardLimitForm';
export * from './OrderCreditCardBilliable';
export * from './OrderDebitSmartCard';
export * from './TransferByPhoneNumberForm';
export * from './AcceptTerms';
