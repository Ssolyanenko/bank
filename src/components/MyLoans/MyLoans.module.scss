@import 'src/styles/variables.module.scss';

.myLoansWrapper {
  width: 849px;
  display: flex;
  margin-left: 170px;
  flex-direction: column;

  .title {
    height: 36px;
    margin-bottom: 40px;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: $bold;
    font-size: 1.5rem;
    line-height: 36px;
    color: $grayDark_6;
  }

  .noLoans {
    width: 859px;
    height: 56px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 16px 32px;
    background: $grayLight_5;

    .link {
      margin: 0 7px;
      text-decoration: underline;
      cursor: pointer;
    }
  }

  .myLoansTable {
    width: 73%;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 0;
    height: 300px;
    border: 1px solid $grayLight_5;
    border-bottom: none;
    border-radius: 6px;
  }

  .myLoanApplicationsWrapper {
    width: 95%;
    margin-top: 64px;

    .loansAppTable {
      border: 1px solid $grayLight_5;
      border-top: none;
      border-radius: 6px;
    }
  }
}
