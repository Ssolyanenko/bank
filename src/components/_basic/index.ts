export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
