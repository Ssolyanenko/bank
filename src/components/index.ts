export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
