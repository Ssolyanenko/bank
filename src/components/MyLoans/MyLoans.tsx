import React, { FC, ReactElement } from 'react';

import { MY_LOANS } from 'constants/myLoans';
import { MY_LOAN_APPLICATIONS } from 'constants/myLoanApplications';
import { MyLoanApplications } from 'components/_basic/MyLoanApplications';
import { LoanItem } from './LoanItem';
import classes from './MyLoans.module.scss';

interface Props {
  isLoans?: boolean;
  isLoanApplications?: boolean;
}

export const MyLoans: FC<Props> = ({ isLoans = true, isLoanApplications = false }): ReactElement => (
  <div className={classes.myLoansWrapper}>
    <h2 className={classes.title}>My Loans</h2>
    {isLoans ? (
      <>
        <div className={classes.myLoansTable}>
          {MY_LOANS.map(({ id, name, periodInMonths, totalSum }) => (
            <LoanItem key={id + name} id={id} loanTitle={name} period={periodInMonths} totalSum={totalSum} />
          ))}
        </div>

        {isLoanApplications && (
          <div className={classes.myLoanApplicationsWrapper}>
            <h2 className={classes.title}>My Loan Applications</h2>
            <div className={classes.loansAppTable}>
              {MY_LOAN_APPLICATIONS.map(({ id, name, periodInMonths, totalSum, status }) => (
                <MyLoanApplications
                  key={id + name}
                  id={id}
                  loanTitle={name}
                  period={periodInMonths}
                  totalSum={totalSum}
                  status={status}
                />
              ))}
            </div>
          </div>
        )}
      </>
    ) : (
      <div className={classes.noLoans}>
        You don't have any loans to display for now. You may check our
        <a className={classes.link} href="#">
          Loan products
        </a>
        and
        <a className={classes.link} href="#">
          Apply for a loan
        </a>
        online
      </div>
    )}
  </div>
);
