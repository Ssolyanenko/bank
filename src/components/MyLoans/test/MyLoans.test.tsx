import React from 'react';
import { shallow } from 'enzyme';

import { MyLoans } from 'components/MyLoans';

describe('<MyLoans />', (): void => {
  test('should make a snapshot', (): void => {
    const wrapper = shallow(<MyLoans />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });
});
