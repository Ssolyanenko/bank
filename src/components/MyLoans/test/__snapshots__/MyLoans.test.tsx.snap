// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyLoans /> should make a snapshot 1`] = `
<div
  className="myLoansWrapper"
>
  <h2
    className="title"
  >
    My Loans
  </h2>
  <React.Fragment>
    <div
      className="myLoansTable"
    >
      <LoanItem
        id={1}
        loanTitle="Personal loan"
        period={60}
        totalSum="30,347.00"
      />
      <LoanItem
        id={2}
        loanTitle="Car loan"
        period={120}
        totalSum="25,745.00"
      />
      <LoanItem
        id={3}
        loanTitle="Wedding loan"
        period={60}
        totalSum="70,339.00"
      />
    </div>
  </React.Fragment>
</div>
`;
